### PR TITLE
Additional CI linters

### PR DESCRIPTION
**Pull request by breed808**
Update `golangci-lint-action` to the latest version.

Add `prealloc` to check for occasions where slices can be pre-allocated.

Add `perfsprint` to check for unneeded uses of `fmt.Sprintf` or uses that can be replaced with faster alternatives.

### CODE REVIEW

**Comment by jkroepke** [comment]
I really like it that we enable more linters.

For my projects, I have inverted the defaults.

Enable all linters, except some. What did you think about this?

https://github.com/jkroepke/openvpn-auth-oauth2/blob/main/.golangci.yaml#L1-L36

**Comment by breed808** [comment]
I've done the same for any private work projects, so I don't mind reversing the logic. I've made this change and enabled a couple other linters.

I'll continue work enabling more linters in future PR(s).

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -98,9 +98,9 @@ jobs:
           go-version-file: 'go.mod'
 
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3
+        uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.55.2
+          version: v1.58
```
Comment (jkroepke) [inline .github/workflows/lint.yml:103]:
```suggestion
          args: --out-format "github-actions,colored-line-number"
          version: v1.58
```

With that, we could gain useful output in just one run and we could avoid a second run.

**Review by jkroepke — APPROVED** [review]
I love more strictness! Great PR!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (breed808) [reply .github/workflows/lint.yml:103]: Done!

**Pull request merged**